storage/statedb/derive_sha.go

```diff
@@ -33,26 +31,23 @@ type DeriveShaOrig struct{}
 func (d DeriveShaOrig) DeriveSha(list types.DerivableList) common.Hash {
 	trie := NewStackTrie(nil)
 	trie.Reset()
-	keybuf := new(bytes.Buffer)
+	var buf []byte
 
 	// StackTrie requires values to be inserted in increasing
 	// hash order, which is not the order that `list` provides
 	// hashes in. This insertion sequence ensures that the
 	// order is correct.
 	for i := 1; i < list.Len() && i <= 0x7f; i++ {
-		keybuf.Reset()
-		rlp.Encode(keybuf, uint(i))
-		trie.Update(keybuf.Bytes(), list.GetRlp(i))
+		buf = rlp.AppendUint64(buf[:0], uint64(i))
+		trie.Update(buf, list.GetRlp(i))
 	}
 	if list.Len() > 0 {
-		keybuf.Reset()
-		rlp.Encode(keybuf, uint(0))
-		trie.Update(keybuf.Bytes(), list.GetRlp(0))
+		buf = rlp.AppendUint64(buf[:0], 0)
+		trie.Update(buf, list.GetRlp(0))
 	}
 	for i := 0x80; i < list.Len(); i++ {
-		keybuf.Reset()
-		rlp.Encode(keybuf, uint(i))
-		trie.Update(keybuf.Bytes(), list.GetRlp(i))
+		buf = rlp.AppendUint64(buf[:0], uint64(i))
+		trie.Update(buf, list.GetRlp(i))
 	}
 	return trie.Hash()
 }
```

Comment:
Seems conflict. You did migrate it to `types` package in the preceding PR, but here not reflected.

Reply:
Yes that's right. For easier review I didn't reflected this change in the previous PR.
I'll resolve conflict after this PR is merged

Reply:
I agree that this PR should be merged first.

Reply:
Okay. thanks.